src/main/java/sqlline/SqlLineOpts.java

```diff
@@ -779,6 +797,23 @@ private String getValidDateTimePatternOrThrow(String dateTimePattern) {
     return dateTimePattern;
   }
 
+  private String getValidConfirmPatternOrThrow(String confirmPattern) {
+    if (confirmPattern == null) {
+      throw new IllegalArgumentException("Confirm pattern is null");
+    }
+    try {
```

Comment:
It would make sense to check if `confirmPattern` is default of the same as current `confirmPattern` value first. In this case no compilation is required.

Reply:
I didn't understand you here. Could you explain again?

Reply:
Current code do this
1. check for null
2. check compilation
3. check if default

I suggest to do this
1. check for null
2. check if new value is the same as old value and do nothing if they are the same
3. check default value
4. check compilation